### PR TITLE
Save PopulationSizeHistory params

### LINKED ISSUE #274: Store PopulationSizeHistory as a json list
It would be nice to store the passed-in PopulationSizeHistory object in the provenance of a tsdated tree sequence. I don't think it is a very large object, right? It just consists of 2 arrays, neither of which is likely to be millions of numbers long. Any reason why not to store it?

### PR DESCRIPTION
**Pull request by hyanwong**
Fixes #274

### CODE REVIEW

**Comment by hyanwong** [comment]
Does this look right @nspope ?

Also pinging @benjeffery , because I have previously wondered about the general strategy for encoding simple objects into the `tskit` provenance information, and this is one possibility, although it does involve allowing multiple types for a parameter (i.e. either a simple object of the correct type or a dictionary of parameters used to instantiate an object of the correct type). The case I was previously stuck on was whether/how to encode a recombination rate map into tsinfer provenance information. That's potentially a much bigger array, though, so maybe not worth it. See https://github.com/tskit-dev/tsinfer/issues/332#issuecomment-867792691

File: tsdate/demography.py [hunk 1]
```diff
@@ -111,6 +120,17 @@ def __init__(self, population_size, time_breaks=None):
         self.coalescent_breaks = coalescent_breaks
         self.coalescent_rate = coalescent_rate
 
+    def as_init_params(self):
```
Comment (nspope) [inline tsdate/demography.py:123]: Maybe `as_dict` for this method name?

**Review by nspope — APPROVED** [review]
LGTM!

File: tsdate/demography.py (hunk 1 above)
Comment (hyanwong) [reply tsdate/demography.py:123]: True. I guess this entirely contains all the info we need, so `as_dict` is correct?

**Pull request merged**